Source/Scene/Model.js

```diff
@@ -2893,11 +2900,46 @@ function createTexture(gltfTexture, model, context) {
     wrapS === TextureWrap.MIRRORED_REPEAT ||
     wrapT === TextureWrap.REPEAT ||
     wrapT === TextureWrap.MIRRORED_REPEAT;
-
+  var npot;
   var tx;
   var source = gltfTexture.image;
 
   if (defined(internalFormat)) {
+    npot =
+      !CesiumMath.isPowerOfTwo(gltfTexture.width) ||
+      !CesiumMath.isPowerOfTwo(gltfTexture.height);
+
+    // Warning to encourage power of 2 texture dimensions with KHR_texture_basisu
+    if (npot && PixelFormat.isCompressedFormat(internalFormat)) {
+      console.warn(
+        "For maximim compatibility, we encourage power of 2 dimensions when using compressed textures in glTF. See the Model.js constructor documentation for more information."
```

Comment:
A similar warning should go in `GltfTextureLoader`

Reply:
The warning should only be printed if the wrap mode is `REPEAT` or `MIRRORED_REPEAT`, since `CLAMP_TO_EDGE` is totally ok.

Also only print the error for WebGL 1. You can check if a WebGL 2 context is used with `context.webgl2`

Reply:
The warning should be a bit more serious and informative. Something like:

"Compressed texture uses REPEAT or MIRRORED_REPEAT texture wrap mode and dimensions are not powers of two. The texture may be rendered incorrectly."